src/components/TimeCardPage/TimeSheet.js

```diff
@@ -49,33 +47,91 @@ export default function Page() {
+        //setTimesheet(testingTimesheetResp)
         apiClient.getUserTimesheets().then(timesheets => {
             setTimesheets(timesheets); 
-            //By Default just render / select the first timesheet for now  
-            if (timesheets.length > 0) {
-                setTimesheet(timesheets[0])
-            }  
-        });  
+            //By Default just render / select the first timesheet for now 
+            currentTimesheetsToDisplay(timesheets, startDate); 
+        });
     }, [])
 
     const processTimesheetChange = (timesheet) => {
         //Adding the time entry to the table 
         apiClient.addTimeEntry(timesheet); 
     }
 
+    const currentTimesheetsToDisplay = (timesheets, currentStartDate) => {
+        // refactor with moment
+        const currentTimesheets = timesheets.filter(sheet => moment.unix(sheet.StartDate).isSame(currentStartDate, 'day'));
+        currentTimesheets.map(sheet => {
+                                if (sheet.TableData.length === 0){
+                                    sheet.TableData = defaultRows;
+                                } 
+                               return sheet;
+                            });
+
+        if (currentTimesheets.length < 1){
+            currentTimesheets.push(testingTimesheetResp); // TODO: change to make correct timesheets for the week
+        }
+
+        if (currentTimesheets.length > 1){
+
+            const totalHoursForEachDay = {};
+
+            // add the days in that stretch to dictionary 
+            // set all to 0
+            // iterate through each sheet and increment accordingly
 
+            for (let start = moment(startDate); start.isBefore(endDate); start.add(1, 'days')){
+                totalHoursForEachDay[start] = 0;
+            }
+
+            currentTimesheets.forEach(sheet => {
+                sheet.TableData.forEach(entry => {
+                    totalHoursForEachDay[moment.unix(entry.StartDate).set({'minutes':0, 'hours':0, 'seconds':0})] += Number(entry.Duration);
+                });
+            });
+
+            const aggregatedRows = Object.entries(totalHoursForEachDay).map(entry =>
+                ({  "StartDate":String(moment(entry[0]).unix()), 
+                    "Duration":String(entry[1]), 
+                    "Comment":{
+                        "AuthorUUID":"XXXX", 
+                        "Type":"Report / Comment, etc", 
+                        "Timestamp":"", 
+                        "Content":":)" 
+                    }
+                }));
+
+            const aggregatedCol = {
+                "UserID":currentTimesheets[0].UserID, 
+                "TimesheetID":22222, 
+                "Company":"Total",
+                "StartDate":moment(startDate).unix(),
+                "Status":"Accepted",
+                "TableData":aggregatedRows
+            };
+
+            currentTimesheets.push(aggregatedCol);
+
+        }
+
+        setCurrentTimesheets(currentTimesheets);
```

Comment:
⛏️ Would prefer to name this `currentTimesheets` something like `newCurrentTimesheets` just because on first glance it might be confused with the state `currentTimesheets`